weaviate/batch/crud_batch.py

```diff
@@ -717,13 +918,7 @@ def flush(self) -> None:
         if one is provided. (See the docs for `configure` or `__call__` for how to set one.)
         """
 
-        result_objects = self.create_objects()
-        result_references = self.create_references()
-        if self._callback is not None:
-            if result_objects:
-                self._callback(result_objects)
-            if result_references:
-                self._callback(result_references)
+        self._send_batch_requests(force_wait=True)
```

Comment:
Can we check here if the batch is still active? (eg the threadpool not shtudown?)
That was one of the problems with the chaos pipelines that it was called outside of a with block

Reply:
Maybe we do not need to shutdown the ThreadExecutor and leave it up to the user? So basically I ca remove it from here: https://github.com/semi-technologies/weaviate-python-client/blob/multi-threading-batch/weaviate/batch/crud_batch.py#L1300

What do you think?

Reply:
Would the user to then manually shut it down? Or would it happen when the client exits?
I think the two possibilities are:

- Have it in the batch, then we need to remove it when the batch is done (eg when the with statement is exited) to make sure that users do not start 20 batches and have an Threadpool every time
- Have it in the client, then we know that there is only one Threadpool and we can shut it down when the client exits

I think it is fine to leave it at the batch level, we just need to give a proper error message

Reply:
I thought about these 2 options too, and both have pros and cons.

Pros and cons for the first:
- Pro: User gets resources back.
- Cons: User has to start it if he wants to use `batch` after a `with` statement. (I do not think it happens that often)

Pros and Cons for the second:
- Pros: Easy to fix now.
- Cons: Resources never get released, even if you do not want to do an import.

There is also another problem: The Thread Executor gets created at Client init, so if a user want to do only queries, the resources are not released. So now the question is do we want to introduce breaking changes. I would do it like this: Do not create an Executor unless the batch is used with `with` or a user explicitly calls `start` `shutdown`.

Let me know what do you think.

Reply:
I think that the first one is better then.
If you use `with batch...` the first time, we can expect you to do it the second time, too

Reply:
I think I fixed it